webbpsf/optics.py

```diff
@@ -1111,8 +1117,42 @@ def __init__(self, instrument, include_oversize=False, **kwargs):
 
             # Cubic interpolation of of non-uniform 2D grid
             cf = griddata((v2, v3), zvals, (v2_tel, v3_tel), method='cubic').tolist()
+            # Want to perform extrapolation if outside of bounds
             if np.isnan(cf):
-                cf = self.row[zkey]
+                #cf = self.row[zkey]
+
+                # Create fine mesh grid
+                dstep = 1. / 60. # 1" steps
+                xgrid = np.arange(v2.min(), v2.max()+dstep, dstep)
+                ygrid = np.arange(v3.min(), v3.max()+dstep, dstep)
+                X, Y = np.meshgrid(xgrid,ygrid)
+
+                # Cubic interpolation of all points
+                zgrid = griddata((v2, v3), zvals, (X, Y), method='cubic')
+    
+                # There will be some NaNs along the border that need to be replaced
+                ind_nan = np.isnan(zgrid)
+                # Remove rows/cols 1 by 1 until no NaNs
+                zgrid2 = zgrid
+                ygrid2 = ygrid
+                xgrid2 = xgrid
+                while np.isnan(zgrid2.sum()):
+                    zgrid = zgrid2[1:-1,1:-1]
+                    ygrid = ygrid2[1:-1]
+                    xgrid = xgrid2[1:-1]
+
+                # Create regular grid interpolator function for extrapolation of NaN's
+                func = RegularGridInterpolator((ygrid2,xgrid2), zgrid2, method='linear',
```

Comment:
`RegularGridInterpolator` is never imported

Reply:
Good catch. I thought I copied over all the code changes into the new branch, but apparently missed a couple things. My github pull request workflow is still a little awkward for contributed projects.

Reply:
@JarronL I'd be happy to chat sometime re pull requests and branches if you want. There's about a thousand possible workflows with git but I can tell you what I use (which is actually a couple different options, in different repos)